src/librustc/traits/error_reporting.rs

```diff
@@ -392,12 +392,16 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
             return;
         }
 
-        let end = cmp::min(4, impl_candidates.len());
+        let end = if impl_candidates.len() == 5 {
+            5
+        } else {
+            cmp::min(4, impl_candidates.len())
+        };
```

Comment:
How about

```Rust
        let end = if impl_candidates.len() <= 5 {
            impl_candidates.len()
        } else {
            4
        };
```
? At least for me, it would make it easier to immediately grasp the behaviour.